src/main/resources/templates/c_plus_plus/test/tests/TestCatch2.py

```
@@ -0,0 +1,36 @@
+
+    def _run(self) -> None:
+        # Start the program:
+        outputFilename = f"result-{self.executable}.xml"
+        self.pWrap = self._createPWrap([join(".", self.executable), "--success", "--reporter", f"JUnit::out={outputFilename}", "--reporter", "console::out=-::colour-mode=none"], self.executionDirectory)
+        self._startPWrap(self.pWrap)
+        self.pWrap.waitUntilTerminationReading()
+
+        retCode: int = self.pWrap.getReturnCode()
+        # parse XML output and append it to the results
+        try:
+            catchXmlRoot: Et = Et.parse(join(self.executionDirectory, outputFilename))
```

Ruff

Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)